distributed/tests/test_as_completed.py

```diff
@@ -168,3 +170,39 @@ def test_async_for_py2_equivalent(c, s, a, b):
 
     with pytest.raises(StopAsyncIteration):
         yield seq.__anext__()
+
+
+def test_as_completed_error(loop):
+    with cluster() as (s, [a, b]):
+        with Client(s['address'], loop=loop) as c:
+            x = c.submit(throws, 1)
+            y = c.submit(sleep, 1)
+
+            ac = as_completed([x, y])
+            result = list(ac)
+
+            assert result == [x, y]
+            assert x.status == 'error'
+            assert y.status == 'finished'
+
+
+def test_as_completed_with_results(loop):
+    with cluster() as (s, [a, b]):
+        with Client(s['address'], loop=loop) as c:
+            x = c.submit(throws, 1)
+            y = c.submit(sleep, 1)
+            z = c.submit(inc, 1)
+
+            ac = as_completed([x, y, z], with_results=True)
+            y.cancel()
+            res = list(ac)
+
+            futs, results = zip(*res)
+            assert futs == (y, x, z)
```

Comment:
How confident are we of this ordering?  We might also pick out x, y, and z based on their keys.

Reply:
Good question - I can update so it's order independent.